From 60ff7fbcf1cf4f558a26bf990296e15199cf4b86 Mon Sep 17 00:00:00 2001
From: Sergey Volkov <Sergey.Volkov@promwad.com>
Date: Sat, 19 Feb 2022 15:12:57 +0300
Subject: [PATCH 9/9] rzx27: disable all default biases for saving some power

There were a lot of enabled pull-ups at almost all pins.
An internal pull-ups are unnecessary in 99% of cases, here too.
---
 src/board-rzx27.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/board-rzx27.c b/src/board-rzx27.c
index a303a05..d8b0b28 100644
--- a/src/board-rzx27.c
+++ b/src/board-rzx27.c
@@ -437,6 +437,12 @@ void board_init(void)
 	/* MSC1 pins */
 	__gpio_as_func_mask(GPIOD, 0x1c000000, 0);
 
+	/* disable all biases */
+	__gpio_disable_pull_mask(0, 0xffffffff);
+	__gpio_disable_pull_mask(1, 0xffffffff);
+	__gpio_disable_pull_mask(2, 0xffffffff);
+	__gpio_disable_pull_mask(3, 0xffffffff);
+
 #ifdef USE_SERIAL
 	__gpio_as_func(2, 12, 1); /* UART_TX */
 
-- 
2.35.1

